static/js/settings_account.js

```diff
@@ -623,14 +588,60 @@ export function set_up() {
                             rendered_error_msg = error_last_user;
                         }
                     }
-                    overlays.close_modal("#deactivate_self_modal");
+                    confirm_dialog.hide_confirm_dialog_spinner();
+                    overlays.close_modal("#confirm_dialog_modal");
                     $("#account-settings-status")
                         .addClass("alert-error")
                         .html(rendered_error_msg)
                         .show();
                 },
             });
         }, 5000);
+    }
+
+    $("#user_deactivate_account_button").on("click", (e) => {
+        // This click event must not get propagated to parent container otherwise the modal
+        // will not show up because of a call to `close_active_modal` in `settings.js`.
+        e.preventDefault();
+        e.stopPropagation();
+
+        const modal_parent = $("#account-settings");
+        const html_body = user_deactivation_modal();
+
+        confirm_dialog.launch({
+            parent: modal_parent,
+            html_heading: $t_html({defaultMessage: "Deactivate your account?"}),
+            html_body,
+            html_yes_button: $t_html({defaultMessage: "Confirm"}),
+            help_link: "/help/deactivate-your-account",
+            on_click: do_deactivate_user_account,
+            fade: true,
+            loading_spinner: true,
+        });
+    });
+
+    $("#account-settings").on("click", ".custom_user_field .remove_date", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        const field = $(e.target).closest(".custom_user_field").expectOne();
+        const field_id = Number.parseInt($(field).attr("data-field-id"), 10);
+        update_user_custom_profile_fields([field_id], channel.del);
+    });
+
+    $("#account-settings").on("change", ".custom_user_field_value", function (e) {
+        const fields = [];
+        const value = $(this).val();
+        const field_id = Number.parseInt(
+            $(e.target).closest(".custom_user_field").attr("data-field-id"),
+            10,
+        );
+        if (value) {
+            fields.push({id: field_id, value});
+            update_user_custom_profile_fields(fields, channel.patch);
+        } else {
+            fields.push(field_id);
+            update_user_custom_profile_fields(fields, channel.del);
+        }
```

Comment:
Can you avoid moving the `custom_user_field` logic, to keep the diff size small?